docs/core.md

```diff
@@ -0,0 +1,122 @@
+
+```C
+  name = PyUnicode_FromString(name_utf8);
+  QUIT_IF_NULL(name);
+  sys_modules = PyImport_GetModuleDict(); // returns borrow
+  Py_XINCREF(sys_modules);
+  QUIT_IF_NULL(sys_modules);
+  module = PyDict_GetItemWithError(sys_modules, name); // returns borrow
+  Py_XINCREF(module);
+  if(module && !JsImport_Check(module)){
+    PyErr_Format(PyExc_KeyError,
+      "Cannot mount with name '%s': there is an existing module by this name that was not mounted with 'pyodide.mountPackage'."
+      , name
+    );
+    QUIT();
+  }
+  QUIT_IF_ERR_OCCURRED();
+// ... [SNIP]
+```
+
+4. The finally block. Here we will clear all the variables we declared at the top in exactly the same order. Do not clear the arguments! They are borrowed. According to the standard Python function calling convention, they are the responsibility of the calling code.
+```
+  success = true;
+finally:
+  Py_CLEAR(sys_modules);
+  Py_CLEAR(importlib_machinery);
+  Py_CLEAR(ModuleSpec);
+  Py_CLEAR(spec);
+  Py_CLEAR(__dir__);
+  Py_CLEAR(module_dict);
+  if(!success){
+    Py_CLEAR(result);
+  }
+  return result;
+}
+```
+
+
+## CPython APIs
+
+### Conventions for indicating errors
+The two main ways to indicate errors:
+1. If the function returns a pointer, (usually PyObject*) then to indicate an error set an exception and return NULL.
+2. If the function returns int and the output is nonnegative, to indicate an error set an exception and return NULL.
+
+Certain functions have "successful errors" like ``PyIter_Next`` (successful error is "StopIteration") and ``PyDict_GetItemWithError`` (successful error is "KeyError"). These functions will return NULL without setting an exception to indicate the "successful error" occurred. Check what happened with ``PyErr_Occurred``. Also, functions that return ``int`` for which ``-1`` is a valid return value will return ``-1`` with no error set to indicate that the result is ``-1`` and ``-1`` with an error set if an error did occur. The simplest way to handle this is to always check ``PyErr_Occurred``.
+
+Lastly, the argument parsing functions ``PyArg_ParseTuple``, ``PyArg_Parse``, etc behave weirdly. These return ``true`` on success and return ``false`` and set an error on failure.
```

Comment:
"behave weirdly" is not OK wording wise for the documentation.

Reply:
Changed to "are edge cases"